### PR TITLE
BackupSchedules api draft

### PR DESCRIPTION
**Pull request by qrort**

### CODE REVIEW

File: pkg/proto/backup_schedule_service.proto [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+package ydbcp;
+option go_package = "github.com/ydb-platform/ydbcp/pkg/proto;ydbcp";
+
+import "backup_schedule.proto";
+import "google/protobuf/duration.proto";
+
+service BackupScheduleService {
+    rpc CreateBackupSchedule (CreateBackupScheduleRequest) returns (BackupSchedule);
+    rpc UpdateBackupSchedule (UpdateBackupScheduleRequest) returns (BackupSchedule);
```
Comment (qrort) [inline pkg/proto/backup_schedule_service.proto:11]: in old API, there is no Create\Update, only Backup (which basically means SetBackupScheduleSettingsForDatabase)

File: pkg/proto/backup_schedule_service.proto [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    string next_page_token = 2;
+}
+
+//QUESTION: do we need UpdateBackupScheduleRequest?
+//it is almost the same as create.
+message UpdateBackupScheduleRequest {
+    string id = 1;
+    //QUESTION: or maybe we leave out all of this and
+    //allow to change only BackupScheduleSettings and name
+    string container_id = 2;
```
Comment (bma13) [inline pkg/proto/backup_schedule_service.proto:61]: for what we need container_id?

File: pkg/proto/backup_schedule.proto [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+message BackupSchedulePattern {
+    //oneof?
+    BackupSchedulePatternOption pattern = 1;
+    string crontab = 2;
+}
+
+message BackupScheduleSettings {
+    BackupSchedulePattern schedule_pattern = 1;
+    google.protobuf.Duration ttl = 2;
```
[the remark this answers is not in the document]
Comment (bma13) [reply pkg/proto/backup_schedule.proto:25]: > If we have ttl for each schedule, then we need to have optional ScheduleId field in Backup entity
I think we should use this ttl value when creating a backup and fill expire_at field

btw store ScheduleId in Backup record can be useful

File: pkg/proto/backup_schedule.proto [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    float last_backup_rpo_margin_percent = 4;
+}
+
+message BackupSchedule {
+    //backup settings
+    string id = 1;
+    string container_id = 2;
+    string database_name = 3;
+    string endpoint = 4;
+    repeated string source_paths = 5;  // [(size) = "<=256"];
```
[the remark this answers is not in the document]
Comment (qrort) [reply pkg/proto/backup_schedule.proto:42]: https://github.com/ydb-platform/ydb/pull/7743 check for > 256 paths passes. I'd suggest we leave it here for now and wait until someone from right verifies it is ok

File: pkg/proto/backup_schedule_service.proto (hunk 2 above)
Comment (qrort) [reply pkg/proto/backup_schedule_service.proto]: to know in which container we should create backups

File: pkg/proto/backup_schedule_service.proto (hunk 1 above)
Comment (qrort) [reply pkg/proto/backup_schedule_service.proto]: I settled on leaving only `rpc SetBackupSchedule`

File: pkg/proto/ydbcp/v1alpha1/backup_schedule.proto [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    ydbcp.v1alpha1.AuditInfo audit = 7;
+
+    string schedule_name = 8;
+    bool active = 9;
+
+    BackupScheduleSettings schedule_settings = 10;
+
+    google.protobuf.Timestamp next_launch = 11;
+
+    ScheduledBackupInfo last_successful_backup_info = 12;
```
Comment (qrort) [inline pkg/proto/ydbcp/v1alpha1/backup_schedule.proto:46]: Maybe we also need last_backup_id or last backup info. To determine if we need to launch first backup.

**Review by robdrynkin — APPROVED** [review]

**Review by bma13 — APPROVED** [review]

**Pull request merged**